### DISCUSSION TITLE
no tray icon after reboot

### DISCUSSION THREAD

**Discussion by hp8wvvvgnj6asjm7** [rustdesk/rustdesk#8636] [Q&A] [answered]
### Bug Description

There is no tray icon after installing the service and rebooting the machine.
The service is enabled, and is also running in the background.

NOTE: The icon is visible before restart!
Therefore it is probably not an issue with Windows itself.

Also is it normal to see two `rustdesk.exe`?

### How to Reproduce

Install service and reboot.

### Expected Behavior

Show icon.

### Operating system(s) on local side and remote side

Windows 10

### RustDesk Version(s) on local side and remote side

1.2.7

### Screenshots

![image](https://github.com/rustdesk/rustdesk/assets/45887282/1c2b67c1-2e4b-441d-abdb-5ec0375d2dcd)

**Comment by z0ow** [comment]
I have the same problem as you but the system is Windows 11 23h2. The RustDesk service is normally up but no icon is shown in the tray.

**Reply by hp8wvvvgnj6asjm7** [reply]
why did the devs move this to discussion? it is clearly a bug, not a feature.

**Comment by 21pages** [comment]
@hp8wvvvgnj6asjm7 @z0ow 

1. How did you install rustdesk?
2. After reboot, is there a new log file in `%appdata%/RustDesk/log/tray`? You can test by deleting all tray log files and reboot.
3. Does RustDesk Tray shortcut exist in directory `C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Startup\` ?

**Reply by z0ow** [reply]
> @hp8wvvvgnj6asjm7 @z0ow
> 
> 1. How did you install rustdesk?
> 2. After reboot, is there a new log file in `%appdata%/RustDesk/log/tray`? You can test by deleting all tray log files and reboot.
> 3. Does RustDesk Tray shortcut exist in directory `C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Startup\` ?

> @hp8wvvvgnj6asjm7 @z0ow
> 
> 1. How did you install rustdesk?
> 2. After reboot, is there a new log file in `%appdata%/RustDesk/log/tray`?
> 3. Does RustDesk Tray shortcut exist in directory `C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Startup\` ?

Honest thank you for tracking this issue.

1. The RustDesk is locally installed by the binary download from the release page, and the installation is located at `C:\Program Files\RustDesk\`
2. The `RustDesk_rCURRENT.log` is updated only when manually opening RustDesk.lnk from either the desktop or startup menu, so it is not changed after the shutdown->power on or reboot (I am guessing this may be a cue)
3. Yes, and it is linked to target `"C:\Program Files\RustDesk\rustdesk.exe" --tray`
4. Even though the icon is not in the tray, I have reconfirmed again that the RustDesk service (`C:\Program Files\RustDesk\RustDesk.exe" --service`) status is running

Then, I checked the Strat Application in the Task Manager it is "not allowed". Hence, I have changed the status to "allowed" and the tray icon is returned as expected.

Overall, this is not a bug but the launch logic confuses me in the following aspects:
1. Whatever the Startup behavior of RustDesk is the RustDesk service is always enabled. Which means I do not have the power to control whether to enable or disable it.
2. The RustDesk is always running as a service but can not show the icon in the tray.

So, starting the RustDesk service and showing the tray both by the RustDesk.lnk could be a better way?

**Reply by rustdesk** [reply, marked as answer]
> Then, I checked the Strat Application in the Task Manager it is "not allowed". Hence, I have changed the status to "allowed" and the tray icon is returned as expected.
